src/main/java/game/map/Map.java

```diff
@@ -77,4 +80,42 @@ public List<Player> getPlayers() {
         return players;
     }
 
-}
+        public static HashMap<Player, List<Country>> assignCountries(List<Player> players, List<Country> countries) {
+            HashMap<Player, List<Country>> playerCountryMap = new HashMap<>();
+
+            // Check if the countries list is empty
+            if (countries.isEmpty()) {
+                System.out.println("No countries available to assign.");
+                return playerCountryMap;
+            }
+
+            // Shuffle the countries
+            Collections.shuffle(countries);
+
+            // Calculate the number of countries each player should have
+            int numCountriesPerPlayer = 5;
+
+            // Assign countries to players
+            int currentIndex = 0;
+            for (Player player : players) {
+                List<Country> assignedCountries = new ArrayList<>();
+                for (int i = 0; i < numCountriesPerPlayer && currentIndex < countries.size(); i++) {
```

Comment:
I think we can just change the for loop to `for (int i = 0; i < numCountriesPerPlayer; i ++)`. This should not cause any damage, if you update `numCountriesPerPlayer` as I suggested before.

Reply:
completed